gazebo_ros/test/entity_spawner.test.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import ament_index_python
+
+import launch
+import launch.actions
+
+from launch.actions import IncludeLaunchDescription
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+import launch_testing
+import launch_testing.asserts
+import launch_testing.markers
+import launch_testing.tools
+import pytest
+
+
+@pytest.mark.launch_test
+@launch_testing.markers.keep_alive
+def generate_test_description(ready_fn):
+    return launch.LaunchDescription([
+        launch.actions.DeclareLaunchArgument(
+            'mock_robot_state_publisher',
+            description='Path to mock robot state publisher executable',
+        ),
+        launch.actions.ExecuteProcess(
+            cmd=[launch.substitutions.LaunchConfiguration('mock_robot_state_publisher')],
+            output='screen'
+        ),
+        IncludeLaunchDescription(
+            PythonLaunchDescriptionSource([
+                ament_index_python.get_package_share_directory('gazebo_ros'),
+                '/launch', '/gzserver.launch.py'
+                ])
+        ),
+        launch.actions.OpaqueFunction(function=lambda context: ready_fn())
+        # launch_testing.actions.ReadyToTest()
```

Comment:
Can the comment be removed?

Reply:
I am waiting on @hidmic to comment on this. `ReadyToTest` is apparently the preferred way of handling this, but it yields to problems on my machine. So I hope to get this solved.

Reply:
It is the preferred way indeed, and thus I'm curious why it doesn't work for you.

Reply:
The tests fails with output similar to this:

```
19: [INFO] [sh-1]: process started with pid [74368]
19: Timed out waiting for processes to start up
19: [INFO] [sh-1]: sending signal 'SIGINT' to process[sh-1]
```
```
19: [INFO] [gzserver-2]: process has finished cleanly [pid 73832]
19: Processes under test stopped before tests completed
19: -- run_test.py: return code 1
```

The test run correctly when using the _old_ method.

Reply:
@Karsten1987 did you remove `ready_fn` from the arguments' list, as in [examples/ready_action_test.py](https://github.com/ros2/launch/blob/a2f3255aa7a9823fbf330269f170d1350d8a0378/launch_testing/test/launch_testing/examples/ready_action_test.py) when you tried `launch_testing.actions.ReadyToTest`? On a probably unpleasant note, trying to use both old and new methods simultaneously may leave you in a state where neither works (see [here](https://github.com/ros2/launch/blob/f73bc56caa9c7b3c53b264475728392d0e1ce9cf/launch_testing/launch_testing/loader.py#L37-L69)).

Reply:
That seems to work. See https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1033/commits/880cf8bb9f174c95d0aa5626a43bfebeb40105be

Is there a way to warn the user of this UB? Something like issuing a warning if `ready_fn` is available but the `OpaqueFunction` isn't used? I don't know the implications of this in the first place, just thinking out loud.